spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/mkldnn/DnnGraph.scala

```diff
@@ -153,6 +163,57 @@ class DnnGraph(
     this
   }
 
+  /**
+   * For some blas layer, we may not have to compute its forward primitives
+   */
+  private def skipInitFwdPrimitives() : Unit = {
+    val skipNodesMap = new mutable.HashMap[String, Boolean]()
+    var i = forwardExecution.length - 1
+    while (i >= 0) {
+      val node = forwardExecution(i)
+      if (this.train) {
+        skipPrimitiveId(i) = false
+      } else {
+        skipPrimitiveId(i) = skip(node, skipNodesMap)
+        skipNodesMap(node.element.getName()) = skipPrimitiveId(i)
+      }
+      i -= 1
+    }
```

Comment:
does it mean  only works at inference ? if is, I think we should not iterate whole executions at training.

Reply:
yeah, here can be some code refactor.